src/main/java/com/okta/tools/authentication/OktaAuthentication.java

```diff
@@ -0,0 +1,136 @@
+package com.okta.tools.authentication;
+
+import com.okta.tools.OktaAwsCliEnvironment;
+import com.okta.tools.models.AuthResult;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONObject;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+
+public final class OktaAuthentication {
```

Comment:
👍 I like this. Moving authentication into it's own class is really good.

Reply:
The tool used to reprompt if a bad username/password is inputted, but now gives this error instead, and crashes.
Exception in thread "main" java.lang.RuntimeException: Failed : HTTP error code : 401
	at com.okta.tools.authentication.OktaAuthentication.primaryAuthFailureHandler(OktaAuthentication.java:106)
	at com.okta.tools.authentication.OktaAuthentication.getPrimaryAuthResponse(OktaAuthentication.java:51)
	at com.okta.tools.authentication.OktaAuthentication.getOktaSessionToken(OktaAuthentication.java:30)
	at com.okta.tools.saml.OktaSaml.getSamlResponse(OktaSaml.java:33)
	at com.okta.tools.OktaAwsCliAssumeRole.run(OktaAwsCliAssumeRole.java:62)
	at com.okta.tools.WithOkta.main(WithOkta.java:23)
15:03:33 [main] ERROR com.okta.tools.authentication.OktaAuthentication - Invalid username or password.